src/ConstructiveSolidGeometry/SurfacePrimitives/Plane.jl

```diff
@@ -1,168 +1,37 @@
-    ]
-end
 
-function Quadrilateral(;p1 = CartesianPoint{Float32}(0,0,0), p2 = CartesianPoint{Float32}(1,0,0), p3 = CartesianPoint{Float32}(1,1,0), p4 = CartesianPoint{Float32}(0,1,0))
-    T = float(promote_type(eltype.((p1, p2, p3))...))
-    tri = Triangle(p1, p2, p3)
-    #will return triangle if conditions are not met
-    if geom_round(p4) in [geom_round(p1), geom_round(p2), geom_round(p3)]
-        return tri
-    else
-        if on_infinite_plane(p4, tri)
-            if !projection_in_triangle(p4, tri)
-                v = CartesianVector{T}(p4 - p1)
-                v2 = CartesianVector{T}(p3 - p1)
-                n = get_surface_vector_nonunitary(tri)
-                if geom_round(dot(n,cross(v2,v))) ≥ 0
-                    return Plane(T, p1, p2, p3, p4)
-                else
-                    return tri
-                end
-            else
-                return Plane(T, p1, p2, p3, p4)
-            end
-        else
-            return tri
-        end
-    end
+"""
+    We assume l.direction is normalized
+"""
+function intersection(p::Plane{T}, line::Line{T}) where {T}
+    ndir = normalize(line.direction)
+    λ = (p.normal ⋅ p.origin - p.normal ⋅ line.origin) / (p.normal ⋅ ndir)
+    line.origin + λ * ndir
 end
 
-Quadrilateral(p1, p2, p3, p4) = Quadrilateral(;p1 = p1, p2 = p2, p3 = p3, p4 = p4)
-Plane(p1, p2, p3, p4) = Quadrilateral(;p1 = p1, p2 = p2, p3 = p3, p4 = p4)
-Plane(p1, p2, p3) = Triangle(;p1 = p1, p2 = p2, p3 = p3)
-
-get_vertices(quad::Plane{T, CartesianPoint{T}, CartesianPoint{T}}) where {T} = [quad.p1, quad.p2, quad.p3, quad.p4]
+function intersection_with_φ_axis(p::Plane{T}, r::T, z::T) where {T}
+    n = normal(p)
+    o = origin(p)
+    u = n × CartesianVector{T}(-n[3], n[1], n[2])
+    v = n × u
 
+    error("This will be fun...")
 
-function decompose_into_tiangles(quad::Plane{T, CartesianPoint{T}, CartesianPoint{T}})::Tuple{Plane{T, CartesianPoint{T}, Nothing},Plane{T, CartesianPoint{T}, Nothing}} where {T}
-    quad.p4 in Triangle(quad.p2,quad.p1,quad.p3) || quad.p2 in Triangle(quad.p4,quad.p1,quad.p3) ? (Triangle(quad.p1,quad.p2,quad.p4), Triangle(quad.p3,quad.p2,quad.p4)) : (Triangle(quad.p2,quad.p1,quad.p3), Triangle(quad.p4,quad.p1,quad.p3))
-end
-
-
-function in(point::CartesianPoint{T}, quad::Plane{T, CartesianPoint{T}, CartesianPoint{T}})::Bool where {T}
-    tri1, tri2 = decompose_into_tiangles(quad)
-    point in tri1 || point in tri2
-end
-
-function distance_to_surface(point::CartesianPoint{T}, quad::Plane{T, CartesianPoint{T}, CartesianPoint{T}})::T where {T}
-    tri1, tri2 = decompose_into_tiangles(quad)
-    min(distance_to_surface(point, tri1), distance_to_surface(point, tri2))
-end
-
-function sample(quad::Plane{T, CartesianPoint{T}, CartesianPoint{T}}, sampling::Union{Real, NTuple{3,Int}}) where {T}
-    tri1, tri2 = decompose_into_tiangles(quad)
-    samples = sample(tri1, sampling)
-    append!(samples, sample(tri2, sampling))
+    n ⋅ u, n ⋅ v
 end
```

Comment:
??

Reply:
I can delete this. Was not sure if would continue implementing it at some point.